src/Symfony/Component/String/AbstractString.php

```diff
@@ -620,14 +620,17 @@ public function truncate(int $length, string $ellipsis = '', bool|TruncateMode $
         }
 
         $desiredLength = $length;
-        if (TruncateMode::WordAfter === $cut || TruncateMode::WordBefore === $cut || !$cut) {
+        if (TruncateMode::WordAfter === $cut || !$cut) {
             if (null === $length = $this->indexOf([' ', "\r", "\n", "\t"], ($length ?: 1) - 1)) {
                 return clone $this;
             }
 
             $length += $ellipsisLength;
         }
 
+        if (TruncateMode::WordBefore) {
+        }
```

Comment:
?

Reply:
not complete fix for https://github.com/symfony/symfony/pull/57778#issuecomment-2239294816 🙈